Classes/KIFUITestActor.h

```diff
@@ -477,6 +493,48 @@ static inline KIFDisplacement KIFDisplacementForSwipingInDirection(KIFSwipeDirec
 - (void)swipeAccessibilityElement:(UIAccessibilityElement *)element inView:(UIView *)viewToSwipe inDirection:(KIFSwipeDirection)direction;
 
 /*!
+ @abstract Pulls down on the view that enables the pull to refresh.
+ @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
+ @param label The accessibility label of the view to swipe.
+ */
+- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label;
+
+/*!
+ @abstract Pulls down on the view that enables the pull to refresh.
+ @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
+ @param label The accessibility label of the view to swipe.
+ @param pullDownDuration The enum describing the approximate time for the pull down to travel the entire distance
+ */
+- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label pullDownDuration:(KIFStepperPullToRefreshTiming) pullDownDuration;
```

Comment:
Re: Exposing the pulldown duration.


My thought was that the text string while the pull-down is in motion could be tested. If it's too quick, KIF will likely miss testing for the string in a [tester waitFor...]. 
Hence, this gives the test case author the ability to slow it down enough for KIF to perform the check.


In hindsight, is performing a check like that even possible while KIF is performing the pull-down?
Asked more succinctly, in your opinion, what's the most reliable way to implement the ability to test that?



Thanks,
Mike


-----Original Message-----
From: phatmann <notifications@github.com>
To: kif-framework/KIF <KIF@noreply.github.com>
Cc: mikelupo <mikelupo@aol.com>
Sent: Fri, Sep 25, 2015 6:48 am
Subject: Re: [KIF] Add new pullToRefresh methods with associated label and identifier tests (#722)



In Classes/KIFUITestActor.h:
> @@ -477,6 +493,48 @@ static inline KIFDisplacement KIFDisplacementForSwipingInDirection(KIFSwipeDirec
>  - (void)swipeAccessibilityElement:(UIAccessibilityElement *)element inView:(UIView *)viewToSwipe inDirection:(KIFSwipeDirection)direction;
>  
>  /*!
> + @abstract Pulls down on the view that enables the pull to refresh.
> + @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
> + @param label The accessibility label of the view to swipe.
> + */
> +- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label;
> +
> +/*!
> + @abstract Pulls down on the view that enables the pull to refresh.
> + @discussion This will enact the pull to refresh by pulling down the distance of 1/2 the height of the view found by the accessibility label.
> + @param label The accessibility label of the view to swipe.
> + @param pullDownDuration The enum describing the approximate time for the pull down to travel the entire distance
> + */
> +- (void)pullToRefreshViewWithAccessibilityLabel:(NSString *)label pullDownDuration:(KIFStepperPullToRefreshTiming) pullDownDuration;

Why do we need to expose pull down duration? Do some views require more or less time?
—
Reply to this email directly or view it on GitHub.